File: csharp/src/Training.Workshop/M6/S17Singleton/Step1/TicketDesk.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S17Singleton.Step1;

/// <summary>Krok 1: klient pyta o jedyną instancję zamiast robić new.</summary>
public sealed class TicketDesk
{
    public Money Quote(string format, bool online)
    {
        var price = PriceList.GetInstance().BasePrice(format);
        return online ? price.Plus(Money.Of("2.00")) : price;
    }
}

File: typescript/src/workshop/m6/s17_singleton/step2/TicketDesk.ts
import { Money } from '../../../shared/Money.js';
import { PriceList } from './PriceList.js';

/** Krok 2: klient używa stałej PriceList - wciąż ukryta, globalna zależność. */
export class TicketDesk {
  quote(format: string, online: boolean): Money {
    const price = PriceList.basePrice(format);
    return online ? price.plus(Money.of('2.00')) : price;
  }
}
